### ISSUE TITLE
Headings text colour contrast 

### ISSUE THREAD

**Issue by chris-gds** [tailwindlabs/tailwindui-issues#1510]
**What component (if applicable)**
- URL for category: Accessibility bug
- Component name: Headings

**Describe the bug**
Some of the text on the heading component doesn't pass colour contrast checks.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to: https://tailwindui.com/components/application-ui/headings/page-headings
2. Review colour contrast of text.
3. See error

**Expected behaviour**
**Screenshots**

![screenshot of Tailwind UI heading element with text that fails colour contrast](https://github.com/tailwindlabs/tailwindui-issues/assets/71266765/a93ba441-5a7c-4ff0-82ac-1d69f1a18fcd)

**Additional context**
[Understanding SC 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum)

**Comment by adamwathan** [comment]
Hey! The light text in that screenshot is #6b7280 which has a 4.83:1 contrast ratio against the white background which passes at AA.

![image](https://github.com/tailwindlabs/tailwindui-issues/assets/4323180/0bfd9095-f626-4734-bd63-1d92a3f1fb76)

<img width="776" alt="image" src="https://github.com/tailwindlabs/tailwindui-issues/assets/4323180/60654ad0-be43-4107-a538-8f02a52e2db9">

The #9ca3af color you've pulled there is the color of the icons, which is deliberately a bit lighter than the text to feel optically balanced with the text color, because icons by their nature are heavier than text.

We try hard to be mindful of color contrast guidelines in everything in Tailwind UI, but there will definitely be some places that don't pass where we've made the conscious trade-off to prioritize the visual design over contrast guidelines. One example is the border color on form inputs for instance, which are not as dark as the WCAG contrast guidelines recommend.

I appreciate that for some people this is an unacceptable decision — to that all I can say is that we do try very, very hard to be in the 99th percentile of the internet from an accessibility perspective, and I hope we get credit for the effort we put in that almost no other websites match. Color contrast is the only place where we ever deliberately make any exceptions, because sometimes we need to balance strict guideline adherence with producing visual designs that can compete with the aesthetic of companies like Stripe or Linear, which is what a huge number of our customers are looking for from us.

Hope that's understandable and of course you can tweak anything you want about any of the snippets in Tailwind UI, so if you'd prefer to increase the contrast in your own implementation you can 👍
